### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by maksimok93**

### CODE REVIEW

File: spec/features/02_search_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  it 'Verifies product search without options' do
+    $search.search_product('Xiaomi mi play')
+    expect(page).to have_content(ProductsConstants::XIAOMI)
+  end
+
+  it 'Verifies search without options' do
+    $search.search_product('garmin 235')
+    expect(page).to have_content('Garmin Forerunner 235 Black/Grey (010-03717-55)')
+  end
```
Comment (AVzayats) [inline spec/features/02_search_spec.rb]: what is the difference between these two scenarios?

File: spec/features/02_search_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  it 'Verifies product search with specific price range' do
+    $search.search_product('Xiaomi Redmi')
+    $options.select_price_range(6000, 6400)
+    expect(page).to have_content(ProductsConstants::REDMI)
```
Comment (AVzayats) [inline spec/features/02_search_spec.rb]: please, add price verification as well

File: spec/features/03_cart_spec.rb [hunk 3]
```diff
@@ -32,7 +25,7 @@ def add_watch_to_cart
     end
   end
 
-  describe 'When to Cart added two products' do
+  context 'When to Cart added two products' do
```
Comment (AVzayats) [inline spec/features/03_cart_spec.rb]: better use 'when two products added to Cart'

File: spec/features/04_delete_user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  context 'When on Main Page' do
+    it 'Verifies deleting account' do
+      page.windows[0].maximize
+      visit('/')
+      click_link('войдите в личный кабинет')
+      users_data = eval(File.open('tests_data/users_data', 'r:UTF-8', &:read))
+      auth = Authorization.new(users_data[:email], users_data[:password])
+      auth.auth_as_registered_user
+      click_link('Мои заказы')
+      user = UserActions.new
```
Comment (AVzayats) [inline spec/features/04_delete_user_spec.rb]: it is better to move user creation to before

File: spec/features/02_search_spec.rb (hunk 1 above)
Comment (maksimok93) [reply spec/features/02_search_spec.rb]: This temporary test was added to verify quantity in Test Results. Also, It has been removed.

File: spec/features/02_search_spec.rb (hunk 2 above)
Comment (maksimok93) [reply spec/features/02_search_spec.rb]: Done!

File: spec/features/delete_user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+require_relative '../spec_helper'
+
+RSpec.describe 'Delete user', type: :feature do
+  context 'When on Main Page' do
+    before(:each) do
+      page.windows[0].maximize
+      visit('/')
+      click_link('войдите в личный кабинет')
+      click_link('Зарегистрироваться')
```
Comment (AVzayats) [inline spec/features/delete_user_spec.rb:10]: You can move code related to user creation to a separate method and also use it in other specs (for example in create one)
NOTE: make sure you do not use other general steps like opening page or login in register_user method

Comment (maksimok93) [reply spec/features/delete_user_spec.rb:10]: I think these steps are general and independent from each other and I don't know in which single method I should use them.

**Pull request merged**